docs/source/gateway/index.rst

```diff
@@ -1,6 +1,971 @@
+    )
+    print(response)
+
+.. _gateway_client_api:
+
+Client API
+~~~~~~~~~~
+
+To use the ``MLflowGatewayClient`` API, see the below examples for the available API methods:
+
+1. Initialization
+
+.. code-block:: python
+
+    from mlflow.gateway import MlflowGatewayClient
+
+    gateway_client = MlflowGatewayClient("http://my.gateway:8888")
+
+2. Listing all configured routes on the Gateway:
+
+The :meth:`search_routes() <mlflow.gateway.client.MlflowGatewayClient.search_routes>` method returns a list of all configured and initialized ``Route`` data for the MLflow AI Gateway server.
+
+.. code-block:: python
+
+    routes = gateway_client.search_routes()
+    for route in routes:
+        print(route)
+
+Sensitive configuration data from the server configuration file is not returned.
+
+3. Querying a particular route:
+
+The :meth:`query() <mlflow.gateway.client.MlflowGatewayClient.query>` method submits a query to a configured provider route.
+The data structure you send in the query depends on the route.
+
+.. code-block:: python
+
+    response = gateway_client.query(
+        "chat", {"messages": [{"role": "user", "content": "Tell me a joke about rabbits"}]}
+    )
+    print(response)
+
+
+Further route types will be added in the future.
+
+MLflow Models
+~~~~~~~~~~~~~
+The ``Mlflow AI Gateway`` is not used exclusively for interactive interaction. It can also be employed as a production service for automated jobs.
+The example below demonstrates how to use an AI Gateway server from within a custom ``pyfunc`` implementation.
+
+.. note::
+    The custom ``Model`` shown in the example below is utilizing environment variables for the AI Gateway server's uri. These values can also be set manually within the
+    definition or can be applied via :func:`mlflow.gateway.get_gateway_uri` after the uri has been set. For the example below, the value for ``MLFLOW_GATEWAY_URI`` is
+    ``http://127.0.0.1:5000/``. For an actual deployment use case, this value would be set to the configured and production deployment server.
+
+.. code-block:: python
+
+    import os
+    import pandas as pd
+    import mlflow
+
+
+    class Model(mlflow.pyfunc.PythonModel):
+        # Define the request mechanism by using the REST API
+        def request(self, route, data):
+            import os
+            import requests
+
+            url = os.environ["MLFLOW_GATEWAY_URI"] + f"gateway/{route}/invocations"
+
+            import json
+
+            response = requests.post(
+                url=url,
+                json=data,
+            )
+            response = json.loads(response.text)
+            return response
+
+        def predict(self, ctx, inp):
+            import os
+            import mlflow
+
+            # Query a particular route for batch operations at a given temperature
+            responses = []
+            for _, row in inp.iterrows():
+                prompt = row["prompt"]
```

Comment:
```suggestion
            for prompt in inp["prompt"]:
```

Reply:
The example is about to be completely changed :)